ipaserver/plugins/certmap.py

```diff
@@ -91,6 +91,49 @@
 register = Registry()
 
 
+def check_maprule_is_for_trusted_domain(api_inst, options):
+    """
+    Check that the certmap rule references altSecurityIdentities and
+    associateddomain in options is a trusted domain.
+
+    :param api_inst: API instance
+    :param options: options passed to the command
+                    at least ipacertmapmaprule and
+                    associateddomain options are expected for the check
+
+    :raises: ValidationError if altSecurityIdentities is present in the
+             rule but no Active Directory trusted domain listed in
+             associated domains
+    """
+    is_trusted_domain_required = False
+    maprules = options.get('ipacertmapmaprule')
+    if maprules:
+        for maprule in maprules:
```

Comment:
ipacertmapmaprule is an optional single-valued option => maprules is a string, not a list of string. No need to use this loop, you can just use `if 'altsecurityidentities' in maprules.lower():`
lower() is needed to make a case-insensitive comparison.

Reply:
Thanks, I pushed an update that treats `options.get('ipacertmapmaprule')` as a sting.